test/extended/prometheus/prometheus_builds.go

```diff
@@ -132,37 +113,19 @@ func runQueries(metricTests map[string][]metricTest, oc *exutil.CLI, ns, execPod
 			json.Unmarshal([]byte(contents), &result)
 			metrics := result.Data.Result
 
-			// for each test case, register that one of the returned metrics has the desired labels and value
-			for j, tc := range tcs {
-				if tcs[j].nodata && len(metrics) == 0 {
-					tcs[j].success = true
-					break
-				}
-				for _, sample := range metrics {
-					if labelsWeWant(sample, tc.labels) && valueWeWant(sample, tc) {
-						tcs[j].success = true
-						break
-					}
-				}
+			delete(errsMap, query) // clear out any prior failures
+			if (len(metrics) > 0 && !expected) || (len(metrics) == 0 && expected) {
+				dbg := fmt.Sprintf("promQL query: %s had reported incorrect results: %v", query, metrics)
+				fmt.Fprintf(g.GinkgoWriter, dbg)
+				errsMap[query] = fmt.Errorf(dbg)
 			}
 
-			// now check the results, see if any bad
-			delete(errsMap, query) // clear out any prior faliures
-			for _, tc := range tcs {
-				if !tc.success {
-					dbg := fmt.Sprintf("query %s for tests %#v had results %s", query, tcs, contents)
-					fmt.Fprintf(g.GinkgoWriter, dbg)
-					errsMap[query] = fmt.Errorf(dbg)
-					break
-				}
+			if len(errsMap) == 0 {
+				break
```

Comment:
wouldn't it exit from the for loop after the first successful query execution?

Reply:
nice catch! :+1: